c7n/resources/rds.py

```diff
@@ -1819,3 +1822,61 @@ class resource_type(TypeInfo):
         universal_taggable = object()
 
     augment = universal_augment
+
+
+@filters.register('consecutive-snapshots')
+class ConsecutiveSnapshots(Filter):
+    """Returns instances where number of consective daily snapshots is equal to/or greater than n days.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: rds-daily-snapshot-count
+                resource: rds
+                filters:
+                  - type: consecutive-snapshots
+                    days: 7
+    """
+    schema = type_schema('consecutive-snapshots', days={'type': 'number', 'minimum': 1},
+        required=['days'])
+    permissions = ('rds:DescribeDBSnapshots', 'rds:DescribeDBInstances')
+    annotation = 'c7n:DBSnapshots'
+
+    def process_resource_set(self, client, resources):
+        rds_instances = [r['DBInstanceIdentifier'] for r in resources]
+        paginator = client.get_paginator('describe_db_snapshots')
+        paginator.PAGE_ITERATOR_CLS = RetryPageIterator
+        db_snapshots = paginator.paginate(Filters=[{'Name': 'db-instance-id',
+          'Values': rds_instances}]).build_full_result().get('DBSnapshots', [])
+
+        inst_map = {}
+        for snapshot in db_snapshots:
+            if snapshot['DBInstanceIdentifier'] not in inst_map:
+                inst_map[snapshot['DBInstanceIdentifier']] = []
+            inst_map[snapshot['DBInstanceIdentifier']].append(snapshot)
```

Comment:
nitpicking: we're doing special checks above and below to catch when there are no snapshots (yet) for an instance. using a construct like `inst_map.setdefault(snapshot['DBInstanceIdentifier'], []).append(snapshot)` or defining inst_map as a [defaultdict(list)](https://docs.python.org/3/library/collections.html#collections.defaultdict) can be useful alternatives.

Reply:
my apology.. I have no understanding of this construct so not sure how to implement it

Reply:
read through the documentation and see the utility of the construct .. thanks much !!